app/lib/features/attachments/widgets/attachment_section.dart

```diff
@@ -187,40 +219,20 @@ class FoundAttachmentSectionWidget extends ConsumerWidget {
     );
   }
 
-  Widget generalAttachmentHeader(BuildContext context, WidgetRef ref) {
-    final lang = L10n.of(context);
-    final attachmentTitleTextStyle = Theme.of(context).textTheme.titleSmall;
+  Widget attachmentHeader({
+    required BuildContext context,
+    required String title,
+    required VoidCallback? onTapAdd,
+  }) {
     return Row(
       children: [
-        Text(lang.attachments, style: attachmentTitleTextStyle),
-        const Spacer(),
-        ActerInlineTextButton(
-          onPressed: () => selectAttachment(
-            context: context,
-            onLinkSelected: (title, link) {
-              Navigator.pop(context);
-              return handleAttachmentSelected(
-                context: context,
-                ref: ref,
-                manager: attachmentManager,
-                title: title,
-                link: link,
-                attachmentType: AttachmentType.link,
-                attachments: [],
-              );
-            },
-            onSelected: (files, selectedType) {
-              return handleAttachmentSelected(
-                context: context,
-                ref: ref,
-                manager: attachmentManager,
-                attachments: files,
-                attachmentType: selectedType,
-              );
-            },
+        Expanded(
+          child: Text(
+            title,
+            style: Theme.of(context).textTheme.titleSmall,
           ),
-          child: Text(lang.add),
         ),
+        IconButton(onPressed: onTapAdd, icon: Icon(Icons.add)),
```

Comment:
shouldn't the ability to this be conditioned by `canEdit`? Because if they can't edit the item, we don't want to show the button...

Reply:
Ya true, let me add it. Good notice!